File: Packs/CommonScripts/Scripts/SetAndHandleEmpty/SetAndHandleEmpty.py
import demistomock as demisto
from CommonServerPython import *
from typing import Text


def get_value(value, stringify=False):
    if stringify:
        return str(value)
    elif not value or not isinstance(value, (Text, bytes, bytearray)):
        return value
    else:
        try:
            return json.loads(value)
        except (json.decoder.JSONDecodeError, TypeError):
            return value


def main():
    args = demisto.args()
    value = args.get('value')
    key = args.get('key')
    force = args.get('force') == 'true'
    value = get_value(value, args.get('stringify') == 'true')
    if value or force:
        human_readable = f'Key {key} set'
        context_entry = {key: value}
    else:
        human_readable = 'value is None'
        context_entry = {}

    if args.get('append') == 'false' and context_entry:
        demisto.executeCommand('DeleteContext', {'key': key})

    return_results(CommandResults(
        readable_output=human_readable,
        outputs=context_entry,
    ))


if __name__ in ('__builtin__', 'builtins'):
    main()

File: Packs/CommonScripts/Scripts/SetAndHandleEmpty/SetAndHandleEmpty.js
function getValue(value, stringify = false) {
    if (stringify) {
        if (value === null || value === undefined) {
            return '';
        }
        return String(value);
    } else if (!value || !(typeof value === 'string')) {
        return value;
    } else {
        try {
            return JSON.parse(value);
        } catch (error) {
            return value;
        }
    }
}

function main() {
    const keys = argToList(args.key);
    let value = args.value;
    const force = args.force === 'true';
    value = getValue(value, args.stringify === 'true');

    let results = [];
    for (let i = 0; i < keys.length; i++) {
        let humanReadable = '';
        let contextEntry = {};

        if (value || force) {
            humanReadable = `Key ${keys[i]} set`;
            contextEntry = { [keys[i]]: value };
        } else {
            humanReadable = 'value is None';
        }

        if (args.append === 'false' && Object.keys(contextEntry).length > 0) {
            executeCommand('DeleteContext', { key: keys[i], subplaybook: 'auto' });
        }
        results.push({
            Type: entryTypes.note,
            EntryContext: contextEntry,
            ContentsFormat: formats.json,
            Contents: humanReadable,
            HumanReadable: humanReadable
        });
    };
    return results;
}

try {
    return main();
} catch (error) {
    throw 'Error occurred while running the script:\n' + error;
}
